src/amo/components/SearchForm/index.js

```diff
@@ -88,12 +92,14 @@ export class SearchFormBase extends React.Component {
   }
 
   goToSearch(query) {
-    const { addonType, api, pathname, router } = this.props;
+    const { addonType, api, pathname, router, userAgentInfo } = this.props;
     const filters = { query };
 
     if (addonType) {
       filters.addonType = addonType;
     }
+    filters.operatingSystem = convertOperatingSystemToFilterName(
+      userAgentInfo.os.name);
```

Comment:
It looks to me like we need to DRY this up a bit since we set `filters` in the same way in two places. We need something like `const filters = this.setFilters(newFilters)` that uses values from `newFilters` and falls back to values from `this.props`.

Reply:
Oh, good call. I'll have a look at that.